cmd_fetchurl.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+
+	"github.com/direnv/direnv/sri"
+)
+
+// CmdFetchURL is `direnv fetchurl <URL> [SRI_HASH]`
+var CmdFetchURL = &Cmd{
+	Name:   "fetchurl",
+	Desc:   "Fetches a given URL",
+	Args:   []string{"URL", "SRI_HASH"},
+	Action: actionWithConfig(cmdFetchURL),
+}
+
+func cmdFetchURL(env Env, args []string, config *Config) (err error) {
+	if len(args) < 2 {
+		return fmt.Errorf("missing URL argument")
+	}
+
+	var (
+		algo sri.Algo = sri.SHA256
+		url string
+		sriHash string
+		cacheFile string
+	)
+
+	url = args[1]
+	// validate the SRI hash if it exists
+	if len(args) > 2 {
+		sriHash = args[2]
+		algo, err = sri.GetAlgo(sriHash)
+		if err != nil {
+			return err
+		}
+
+		// shortcut if the cache already has the file
+		cacheFile = filepath.Join(config.CacheDir, "fetchurl", sriHash)
```

Comment:
I see.

Reply:
I would just escape `/` with `-` to avoid `/` in the the filename.

Reply:
thanks, I used `_` instead so it can be distinguished from the `<algo>-<hash>` separator.